server/src/search.rs

```diff
@@ -127,24 +135,33 @@ struct MSFacetDistribution {
     facet: HashMap<String, i32>,
 }
 
-pub async fn do_benchmarked_search(q: String, args: SearchQueryArgs) -> Result<SearchResults> {
+pub async fn do_benchmarked_search(q: String, args: SearchQueryArgs) -> SearchResults {
     let start_time = Instant::now();
 
     let results_sections = execute_search(q, args).await;
 
     let time_ms = start_time.elapsed().as_millis();
-    Ok(SearchResults {
+    SearchResults {
         sections: results_sections,
         time_ms,
-    })
+    }
 }
 
 // size=100 seems to be about 10M
 #[cached(size = 500)]
 async fn execute_search(q: String, args: SearchQueryArgs) -> Vec<SearchResultsSection> {
     let parsed = parse_input_query(q);
+    let sanatised_args = sanitise_args(args);
+    return do_geoentry_search(&parsed.tokens, sanatised_args).await;
+}
 
-    return do_geoentry_search(&parsed.tokens, args).await;
+fn sanitise_args(args: SearchQueryArgs) -> SanitisedSearchQueryArgs {
+    let max_size = u16::MAX as usize;
+    SanitisedSearchQueryArgs {
+        limit_buildings: min(args.limit_buildings.unwrap_or(5), max_size),
+        limit_rooms: min(args.limit_rooms.unwrap_or(5), max_size),
+        limit_all: min(args.limit_all.unwrap_or(20), max_size),
+    }
 }
```

Comment:
honestly, both options are fine by me. 
But sanatising user input is always prefeable.
 
Limiting to 1k seems to be a reasonable maximum, that nobody should hit.
`let max_size = u16::MAX as usize` was a bit much in retrospect.